app/Http/Controllers/ContactController.php

```diff
@@ -37,13 +38,18 @@ public function create()
     public function store(Request $request)
     {
         $request->validate([
-            'first_name'=>'required',
-            'last_name'=>'required',
-            'email'=>'required'
+            'first_name' => 'required',
+            'last_name' => 'required',
+            'email' =>'required'
         ]);
-        $imageName = $request->file('avatar')->getClientOriginalName();
-        $imageName = time().'_'.$imageName ;
-        $request->file('avatar')->move('avatar', $imageName);
+
+        //check if is there image selected
+        $imageName= Config('app.avatar_icon');
+        $avatar = $request->file('avatar');
+        if($avatar){
+            $storagePath= Storage::putFile ('public/avatar/', $request->file('avatar'));
+            $imageName = basename($storagePath);
```

Comment:
@hapo-thomnt file trùng tên thì sao?

Reply:
Hàm putFile này sẽ tự động tạo ra một tên mới duy nhất nên không bị trùng đâu.
link: https://laravel.com/docs/5.8/filesystem
keyword: Automatically generate a unique ID for file name